metadata-ingestion/src/datahub/ingestion/source/cassandra.py

```diff
@@ -0,0 +1,592 @@
+            container_key=keyspace_container_key,
+            name=keyspace_name,
+            sub_types=[DatasetContainerSubTypes.KEYSPACE],
+        )
+
+    def _generate_keyspace_container_key(self, keyspace_name: str) -> ContainerKey:
+        return KeyspaceKey(
+            keyspace=keyspace_name,
+            platform=self.platform,
+            platform_instance=self.config.platform_instance,
+            env=self.config.env,
+        )
+
+    # get all tables for a given keyspace, iterate over them to extract column metadata
+    def _extract_tables_from_keyspace(
+        self, keyspace_name: str
+    ) -> Iterable[MetadataWorkUnit]:
+        tables = self.cassandra_session.execute(GET_TABLES_QUERY, [keyspace_name])
+        tables = sorted(
+            tables,
+            key=lambda t: getattr(
+                t, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"]
+            ),
+        )  # sorted so output is consistent
+        for table in tables:
+            # define the dataset urn for this table to be used downstream
+            table_name: str = getattr(
+                table, CASSANDRA_SYSTEM_SCHEMA_COLUMN_NAMES["table_name"]
+            )
+            dataset_name: str = f"{keyspace_name}.{table_name}"
+            dataset_urn = make_dataset_urn_with_platform_instance(
+                platform=self.platform,
+                name=dataset_name,
+                env=self.config.env,
+                platform_instance=self.config.platform_instance,
+            )
+
+            # 1. Extract columns from table, then construct and emit the schemaMetadata aspect.
+            try:
+                yield from self._extract_columns_from_table(
+                    keyspace_name, table_name, dataset_urn
+                )
+            except Exception as e:
+                logger.warning(
+                    f"Failed to extract columns from table {table_name}", exc_info=True
+                )
+                self.report.report_warning(
+                    "table",
+                    f"Exception while extracting table columns {table_name}: {e}",
+                )
+
+            # 2. Construct and emit the status aspect.
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=StatusClass(removed=False),
+            ).as_workunit()
+
+            # 3. Construct and emit subtype
+            yield MetadataChangeProposalWrapper(
+                entityUrn=dataset_urn,
+                aspect=SubTypesClass(
+                    typeNames=[
+                        DatasetSubTypes.TABLE,
+                    ]
+                ),
+            ).as_workunit()
+
+            # 4. TODO: If useful, we can construct and emit the datasetProperties aspect here.
+            # maybe emit details about the table like bloom_filter_fp_chance, caching, cdc, comment, compaction, compression,  ... max_index_interval ...
+            # custom_properties: Dict[str, str] = {}
```

Comment:
Through previous startups I've been in the practice of highlighting where code can be extended upon with simple TODO comments like these, so we can move quicker now while also reducing the time to extend code if the need for implementing the extra bit arises in the future.
But if best practice is to implement it all up-front I can impl this.